Reorder 廣韻字頭 and correct some 字 & 釋義 (WIP)

syimyuzya opened the pull request:
注意：本 PR 依賴 #8 

- 完整修正廣韻字頭順序：解決了以往互聯網公開版《廣韻》音系資料（主要是韻典和 poem 廣韻字音表）長期存在的縫合問題：已知的這些資料均衍生自有女同車「廣韻檢索 JS」之資料表，其字頭來自《廣韻全字表》而釋義來自「宋本廣韻データ」。然而《廣韻全字表》底本為巾箱本（江蘇教育出版社《宋本廣韻》）而「宋本廣韻データ」底本為澤存堂本（周祖謨《廣韻校本》），兩者收字、字序、字形皆有參差，無法簡單地得到對應關係。
  - src/字序表.csv 即為本次工作主要成果，其完整列出了本項目、poem 字音表、「宋本廣韻データ」、韻典網四種資料的字頭順序對應關係
- 加回 poem 字音表所缺之字：包括澤存堂本失收之字，以及因未入U導致 poem 表未收之字。
- 額外修正一些字頭、釋義：為整理字序時所發現的，仍在進一步收集中。
- 「釋義補充」改為「釋義參照」，不再複讀上字釋義，而是給出「上」（若該字為「上同」「亦同」「古文」等）或「下」（若該字與下一字共用釋義，如「並同」「並古文」等），用以解決 poem 表中「釋義補充」內容不正確的問題。

TODO:

- [x] 四個偽字小韻（597 𤜼、646 𡰝、2021 㶒、3373 𣅝）之字亦標注「當刪」
- [x] 「字頭當刪」一欄似乎大材小用，感覺字頭上可附注的信息不止於此（如據《形聲考》給出個別字的更可靠反切字音等）

syimyuzya committed:
wip: change 釋義補充 to 釋義參照

This is for ease of comparison with the new data

syimyuzya committed:
wip: reorder 廣韻 rows with the new 字序表

TBD:
- Apply patches to _poem_'s data
- Add rows missing in _poem_'s data

syimyuzya committed:
wip: add PUA fixes in patches.csv

syimyuzya committed:
wip: include patches to poem's data

syimyuzya committed:
Reorder 廣韻字頭 and correct some 字 & 釋義

- All entries are reordered according to 澤存堂本
  - This solves a long-standing issue with both 韻典 and 廣韻字音表's
    data: Both data tables combine 字頭 from 廣韻全字表 and 釋義 from
    宋本廣韻データ. However 廣韻全字表 is based on 巾箱本 while
    宋本廣韻データ is based on 澤存堂本, which creates mismatches.

- Entries missing in poem's data are added back
  - This includes characters only representable with IDS, and seveval
    additions from 廣韻校本

- More errors in 字頭 & 釋義 are corrected
  - These were discovered when the new 字序表 was being made, and are
    still WIP.

syimyuzya committed:
Add some notes from 形聲考 in patches.csv

syimyuzya committed:
Improve patches

- Marked 4 invalid 小韻s (597 𤜼, 646 𡰝, 2021 㶒, 3373 𣅝)
- Added more checks on patches.csv

untunt commented:
nk2028 调整过小韵地位或反切的小韵中，《广韵形声考》对字头作了修正的如下：

- 141 𤿎<𢻹>
- 646 𡰝<𡰖>（忽略切韵新韵图「𡰢<sub>匣合四齊平</sub>之訛字」，无需和「𡰢」关联）
- 2991a 𣢝<欦>

这些应该收录。99 鈹小韵下「𤿎」也应一起改为「𢻹」

《广韵形声考》中其他对字头的修正由于不涉及 nk2028 对小韵的调整，可以暂缓收录

syimyuzya committed:
Fix inconsistencies in 字序表 & add ccorrections

New corrections:
- 141 𤿎<𢻹>
- 646 𡰝<𡰖> (小韻 no longer invalid)
- 2991 (2991a) 𣢝<欦>

syimyuzya committed:
Fix: newline in 字序表.csv

Was CRLF, should be LF.

syimyuzya committed:
Check format of 校正字頭 in patches.csv

syimyuzya committed:
Add 韻典 to 字序表.csv

ytenx_流水序 is the 序號 on the character's page (or in the page's URL
after `/kyonh/dzih/`).

syimyuzya committed:
Fix build script

The columns of 字序表.csv has changed, so the load (and check) script in
build.py should be updated accordingly.

syimyuzya committed:
Mention about 韻典's data version

syimyuzya committed:
Add 字頭原貌

字頭原貌 is taken from poem's "字頭-原貌" field (only those marked with
"校" but not with "部件換位" or "調整碼位", as the latter cases are for
equivalent characters), plus our patches.

syimyuzya committed:
Add detailed note about 婔

syimyuzya committed:
Use 槩 instead of 概

For consistency with NK2028's 反切原貌 data.

syimyuzya commented:
As a bonus, the issues related to the old "釋義補充" field presented in #7 are also resolved now. Thanks to @FelisNivalis!

syimyuzya committed:
Update README about 廣韻.csv

syimyuzya committed:
Mention 廣韻形聲考 in README

syimyuzya committed:
Merge split.csv into 小韻表.tsv

Also corrected some outdated 字頭s in 小韻表.tsv.
